src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -129,9 +229,46 @@ private string GenerateJsonStr(EventWrittenEventArgs eventData)
                 json.Add("RelatedActivityId", eventData.RelatedActivityId);
             }
 
-            // Generate string-representation of JSON. Also remove newlines.
-            string jsonString = json.ToString(Newtonsoft.Json.Formatting.None);
-            jsonString = jsonString.Replace("\n", "\\n").Replace("\r", "\\r");
+            string jsonString = "";
+            if (!this.writeToConsole)
+            {
+                // This path supports the legacy regex-based parser in Linux Dedicated.
+                // In the future, we'll be able to remove this and use JSON logging only
+                IList<string> jsonKeys = json.Properties().Select(p => p.Name).ToList();
+                List<string> regexValues = new List<string>();
+
+                foreach (string column in this.orderedRegexCol)
+                {
+                    string val = "";
+                    if (jsonKeys.Contains(column))
```

Comment:
Why not simply use `json.TryGetValue(column, out JToken val)`? That would in theory be an O(1) operation. What you're doing now appears to be O(N) where N is the number of properties in the JSON object (constructing jsonKeys is N, and the Contains operation is another N on top of that). It would also allow you to remove at least two lines of code and a redundant dictionary lookup on line 247.

Reply:
I didn't know that was an option :) . I'll add it to my bag of tricks and use it here